base/applications/rapps/unattended.cpp

```diff
@@ -254,14 +254,23 @@ ParseCmdAndExecute(LPWSTR lpCmdLine, BOOL bIsFirstLaunch, int nCmdShow)
         if ((!hMutex) || (GetLastError() == ERROR_ALREADY_EXISTS))
         {
             /* If already started, find its window */
-            HWND hWindow = FindWindowW(szWindowClass, NULL);
-
-            /* Activate window */
-            ShowWindow(hWindow, SW_SHOWNORMAL);
-            SetForegroundWindow(hWindow);
-            if (bAppwizMode)
-                PostMessage(hWindow, WM_COMMAND, ID_ACTIVATE_APPWIZ, 0);
-            return FALSE;
+            HWND hWindow;
+            for (int wait = 2500, inter = 250; wait > 0; wait -= inter)
```

Comment:
Is that added for-loop also required to make it run properly on MS Windows? Or is that a blessing needed only on ReactOS to make it run reliably?

Reply:
It has nothing to do with ROS vs Windows, it is about speed and quantum because there is a race here.

|Process A|Process B|
| ---- | ---- |
|CreateMutex OK|CreateMutex fails|
|...|FindWindow returns NULL|
|CreateWindow|...|